- MDEV-34565 MariaDB crashes with SIGILL because the OS does not support AVX512
  b68c100076ceed77e0260e2906f0944d5f696503

Index: mysys/crc32/crc32c_x86.cc
--- mysys/crc32/crc32c_x86.cc.orig
+++ mysys/crc32/crc32c_x86.cc
@@ -39,7 +39,7 @@ extern "C" unsigned crc32c_sse42(unsigned crc, const v
 
 constexpr uint32_t cpuid_ecx_SSE42= 1U << 20;
 constexpr uint32_t cpuid_ecx_SSE42_AND_PCLMUL= cpuid_ecx_SSE42 | 1U << 1;
-constexpr uint32_t cpuid_ecx_XSAVE= 1U << 26;
+constexpr uint32_t cpuid_ecx_AVX_AND_XSAVE= 1U << 28 | 1U << 27;
 
 static uint32_t cpuid_ecx()
 {
@@ -395,7 +395,7 @@ static bool os_have_avx512()
 
 static ATTRIBUTE_NOINLINE bool have_vpclmulqdq(uint32_t cpuid_ecx)
 {
-  if (!(cpuid_ecx & cpuid_ecx_XSAVE) || !os_have_avx512())
+  if ((~cpuid_ecx & cpuid_ecx_AVX_AND_XSAVE) || !os_have_avx512())
     return false;
 # ifdef _MSC_VER
   int regs[4];
